spectre_analyses/tasks.py

```diff
@@ -0,0 +1,140 @@
+from contextlib import contextmanager
+from functools import partial
+import json
+import os
+import pickle as pkl
+import shutil
+import sys
+
+import celery
+from celery.utils.log import get_task_logger
+
+import spdata.reader
+import spdata.types as ty
+
+from spectre_analyses.celery import app
+import matlab_hooks as mh
+
+
+FILESYSTEM_ROOT = os.path.abspath(os.sep)
+DATA_ROOT = os.path.join(FILESYSTEM_ROOT, 'data')
+STATUS_PATHS = {
+    'all': FILESYSTEM_ROOT,
+    'done': os.path.join(FILESYSTEM_ROOT, 'data'),
+    'processing': os.path.join(FILESYSTEM_ROOT, 'temp'),
+    'failed': os.path.join(FILESYSTEM_ROOT, 'failed')
+}
+Name = str
+Path = str
+
+
+def _data_path(dataset_name: Name) -> Path:
+    return os.path.join(DATA_ROOT, dataset_name, 'text_data', 'data.txt')
```

Comment:
Restrict this to `DATA_ROOT` to prevent path traversal.

Reply:
Partially accepted. I've updated it to the same scenario we have for upload service. Disallowed characters will be replaced by `_`.